### PR TITLE
Template update for 1.0.4/1.1.1

### PR DESCRIPTION
**Pull request by mlorbetske**
Uses the new package set for ASP.NET 1.0.4/1.1.1
Sets the default to be netcoreapp1.1 where applicable
Carries the test SDK updates
Carries the F# updates

@piotrpMSFT @livarcocc @srivatsn

### CODE REVIEW

**Comment by mlorbetske** [comment]
@dotnet-bot test Fedora24 x64 Release Build please
@dotnet-bot test OpenSUSE42.1 x64 Debug Build please
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Debian8.2 x64 Debug Build please
@dotnet-bot test OpenSUSE13.2 x64 Debug Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Fedora24 x64 Release Build please
@dotnet-bot test OpenSUSE42.1 x64 Debug Build please
@dotnet-bot test Ubuntu16.10 x64 Debug Build please

**Comment by piotrpMSFT** [comment]
@mlorbetske ignore the CIs not marked as `Required`. These are experimental.

**Comment by piotrpMSFT** [comment]
@Mattgertz this was approved during today's shiproom via templating engine issue.

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

File: test/dotnet-build.Tests/GivenDotnetBuildBuildsCsproj.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public void ItRunsWhenRestoringToSpecificPackageDir()
                 .Execute()
                 .Should().Pass();
 
-            var outputDll = Directory.EnumerateFiles(Path.Combine(rootPath, "bin"), "*.dll", SearchOption.AllDirectories).Single();
+            var outputDll = Directory.EnumerateFiles(Path.Combine(rootPath, "bin"), "*.dll", SearchOption.AllDirectories).Single(x => !x.Contains("DiaSymReader"));
```
Comment (piotrpMSFT) [inline test/dotnet-build.Tests/GivenDotnetBuildBuildsCsproj.cs]: Are we not on 1.1.1? is DiaSymReader still content??

File: build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets [hunk 2]
```diff
@@ -88,7 +88,7 @@
 
       <DotNetNew ToolPath="$(Stage2Directory)"
                  TemplateType="Mvc"
-                 TemplateArgs="-au Individual --debug:ephemeral-hive"
+                 TemplateArgs="-au Individual -f netcoreapp1.0 --debug:ephemeral-hive --no-tools"
```
Comment (piotrpMSFT) [inline build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets]: why `--no-tools`?

File: test/dotnet-build.Tests/GivenDotnetBuildBuildsCsproj.cs (hunk 1 above)
Comment (mlorbetske) [inline test/dotnet-build.Tests/GivenDotnetBuildBuildsCsproj.cs]: @piotrpMSFT the SDK to pull up to Microsoft.NETCore.App 1.1.1 implicitly isn't in yet

**Pull request merged**